### ISSUE TITLE
ClientConfig::RegionEndpoint getter does not preserve the current SynchronizationContext 

### ISSUE THREAD

**Issue by Sumo-MBryant** [aws/aws-sdk-net#439]
On AWSSDK.S3 3.2.6-beta using .NET Core 1.0, if one attempts to construct a class derived from ```Amazon.Runtime.ClientConfig``` and then accesses ```RegionEndpoint``` for the first time, the value of ```System.Threading.SynchronizationContext.Current``` is not preserved.

It appears to be set to an instance of ```Amazon.Runtime.Internal.Util.AsyncHelpers.ExclusiveSynchronizationContext```.

This is a problem because there doesn't appear to be a running message loop for that synchronization context as per the code here: https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Internal/Util/_mobile/AsyncHelpers.cs#L75 after the execution of desired task.

This means that any ```await``` that is performed on the thread will stall and never complete.

Here is some code demonstrating this issue:

```C#
using Amazon.S3;
using System.Threading;
using System;

namespace S3SynchronizationContext
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var config = new AmazonS3Config();

            var oldContext = SynchronizationContext.Current;
            var regionEndpoint = config.RegionEndpoint;
            var newContext = SynchronizationContext.Current;

            if (newContext != oldContext)
            {
                throw new Exception("Synchronization Context was changed");
            }
        }
    }
}
```

**Comment by normj** [comment]
Thanks for letting us know. I just replicated the issue and I'm testing a fix.

**Comment by normj** [comment]
The fix for this went out in version 3.2.7 of the SDK. Thanks again for letting us know about the issue.

**Comment by Sumo-MBryant** [comment]
Thank you for the quick resolution!

Confirmed fixed on 3.2.7-beta.
